### PR TITLE
Add protected directories management operations

### PR DESCRIPTION
**Pull request by sibprogrammer**

### CODE REVIEW

File: src/Api/Operator/ProtectedDirectory.php [hunk 1]
```diff
@@ -3,9 +3,113 @@
...
+    public function get($field, $value)
+    {
+        $items = $this->getAll($field, $value);
+        return reset($items);
+    }
+
+    /**
+     * @param string $field
+     * @param integer|string $value
+     * @return Struct\Info[]
```
Comment (xgin) [inline src/Api/Operator/ProtectedDirectory.php]: Struct\DataInfo[] is returned in fact

File: src/Api/Operator/ProtectedDirectory.php [hunk 2]
```diff
@@ -3,9 +3,113 @@
...
+        return $this->_delete($field, $value, 'delete-user');
+    }
+
+    /**
+     * @param $command
+     * @param $field
+     * @param $value
+     * @return \PleskX\Api\XmlResponse
+     */
+    private function _get($command, $field, $value)
```
Comment (mrsombre) [inline src/Api/Operator/ProtectedDirectory.php]: Maybe better to move this to \PleskX\Api\Operator, looks like common method?

File: tests/ProtectedDirectoryTest.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        $protectedDirectory = static::$_client->protectedDirectory()->add('/', static::$webspace->id);
+
+        $user = static::$_client->protectedDirectory()->addUser($protectedDirectory, 'john', 'secret');
+        $result = static::$_client->protectedDirectory()->deleteUser('id', $user->id);
+        $this->assertTrue($result);
+
+        static::$_client->protectedDirectory()->delete('id', $protectedDirectory->id);
+    }
+
```
Comment (mrsombre) [inline tests/ProtectedDirectoryTest.php]: Empty line after class body is not PSR-2 compliant

File: src/Api/Operator/ProtectedDirectory.php [hunk 4]
```diff
@@ -3,9 +3,113 @@
 
 namespace PleskX\Api\Operator;
 
+use PleskX\Api\Struct\ProtectedDirectory as Struct;
+
 class ProtectedDirectory extends \PleskX\Api\Operator
 {
```
Comment (mrsombre) [inline src/Api/Operator/ProtectedDirectory.php]: Empty line before class body is not PSR-2 compliant

File: src/Api/Operator/ProtectedDirectory.php [hunk 5]
```diff
@@ -3,9 +3,113 @@
...
+
+        $filterTag = $getTag->addChild('filter');
+        if (!is_null($field)) {
+            $filterTag->addChild($field, $value);
+        }
+
+        $response = $this->_client->request($packet, \PleskX\Api\Client::RESPONSE_FULL);
+        return $response;
+    }
+
```
Comment (mrsombre) [inline src/Api/Operator/ProtectedDirectory.php]: Empty line after class body is not PSR-2 compliant

File: src/Api/Operator/ProtectedDirectory.php (hunk 2 above)
Comment (sibprogrammer) [reply src/Api/Operator/ProtectedDirectory.php]: Not so common unfortunately due to a lot of minor differences in XML-RPC API itself. But as you can see I did some unification in some other operators for "get" methods and will continue working in this direction.

**Pull request merged**